server/ServerFSM.cpp

```diff
@@ -2931,6 +2941,20 @@ sc::result WaitingForTurnEnd::react(const SaveGameRequest& msg) {
     return discard_event();
 }
 
+void WaitingForTurnEnd::SaveTimedoutHandler(const boost::system::error_code& error) {
+    if (error) {
+        DebugLogger() << "Save timed out cancelled";
+        return;
+    }
+
+    DebugLogger() << "Save timed out.";
+    PlayerConnectionPtr dummy_connection = nullptr;
+    Server().m_fsm->process_event(SaveGameRequest(HostSaveGameInitiateMessage(GetAutoSaveFileName(Server().CurrentTurn())), dummy_connection));
+    m_timeout.expires_from_now(std::chrono::seconds(GetOptionsDB().Get<int>("save.auto.interval")));
```

Comment:
what happens when you pass `0` to `expires_from_now`? I can't tell from the documentation. Is there a reason to make this call if the option is set to 0?

Reply:
Handler called when "save.auto.interval" was more than 0. Is it possible option change its value after the server started?

Reply:
I think client option updates can propagate to servers during a game... Would certainly be preferable in a case like this that that happens.

Reply:
Added check.